packages/@aws-cdk/aws-codebuild/lib/artifacts.ts

```diff
@@ -121,6 +121,14 @@ export interface S3ArtifactsProps extends ArtifactsProps {
    * @default true - output will be encrypted
    */
   readonly encryption?: boolean;
+
+  /**
+   * If this is true, the name defined in the buildspec file will be used for artifacts.
+   * Otherwise, the artifact will use the <name> property.
+   *
+   * @default false - ignore buildspec artifact name
+   */
+  readonly useBuildspecName?: boolean;
```

Comment:
I have 2 comments about this name:

1. We have a class called `BuildSpec`, so I think this should be called `useBuildSpecName`.
2. I wonder whether a name like `useNameFromBuildSpec`  or `useNameInBuildSpec` would be a little more descriptive?

Reply:
I like `useNameFromBuildSpec` - I'll update the code!

Reply:
Done! ed1c25b